src/pages/TimetablePage/components/CustomLecture/index.tsx

```diff
@@ -176,11 +176,31 @@ function CustomLecture({ frameId }: { frameId: string | undefined }) {
     e: { target: { value: string } },
   ) => {
     const { target } = e;
-    const newTimeInfo = {
+    let newTimeInfo = {
       ...timeSpaceComponents[index].time,
       [key]: target?.value,
     };
-    const newTimetableTime = changeToTimetableTime(newTimeInfo);
+
+    let newTimetableTime = changeToTimetableTime(newTimeInfo);
+    // 올바르지 않은 시간을 선택했을 시
+    if (newTimetableTime.length === 0) {
+      const newStartTime = Number(newTimeInfo.startHour.slice(0, 2));
+      const newEndTime = Number(newTimeInfo.endHour.slice(0, 2));
+      if (key.slice(0, 5) === 'start') {
+        newTimeInfo = {
+          ...newTimeInfo,
+          endHour: `${newStartTime + 1}시`,
+          endMinute: newStartTime + 1 === 24 ? '00분' : newTimeInfo.startMinute,
+        };
+      } else {
+        newTimeInfo = {
+          ...newTimeInfo,
+          startHour: newEndTime - 1 < 10 ? '09시' : `${newEndTime - 1}시`,
+          startMinute: newEndTime - 1 < 9 ? '00분' : newTimeInfo.endMinute,
+        };
+      }
+      newTimetableTime = changeToTimetableTime(newTimeInfo);
+    }
```

Comment:
`time` `hour` 변수의 의미가 헷갈리게 작성된 것 같아요... 천천히 읽어보니까 이해할 수 있었어요...
그리고 이 부분은 올바르지 않은 시간을 선택했을 때를 나타내는 로직이니까 따로 함수로 빼도 괜찮을 것 같아요!
기능상 문제는 없어서 우선 approve 하겠습니다~
```suggestion
    let newTimetableTime = changeToTimetableTime(newTimeInfo);
    // 올바르지 않은 시간을 선택했을 시
    if (newTimetableTime.length === 0) {
      const newStartHour = Number(newTimeInfo.startHour.slice(0, 2));
      const newEndHour = Number(newTimeInfo.endHour.slice(0, 2));
      if (key.slice(0, 5) === 'start') {
        newTimeInfo = {
          ...newTimeInfo,
          endHour: `${newStartHour + 1}시`,
          endMinute: newStartHour + 1 === 24 ? '00분' : newTimeInfo.startMinute,
        };
      } else {
        newTimeInfo = {
          ...newTimeInfo,
          startHour: newEndHour - 1 < 10 ? '09시' : `${newEndHour - 1}시`,
          startMinute: newEndHour - 1 < 9 ? '00분' : newTimeInfo.endMinute,
        };
      }
      newTimetableTime = changeToTimetableTime(newTimeInfo);
    }
```

Reply:
그렇네요!! 반영했습니다! 감사합니다! 64285e8dcb1a4d71278227c0703f84207712c245